pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -126,10 +114,11 @@ func (s *Server) initDNSCerts() error {
 				istioGenerated = true
 			}
 		}
+
 		// check if signing key file exists the cert dir and if the istio-generated file
 		// exists (only if USE_CACERTS_FOR_SELF_SIGNED_CA is enabled)
-		if !detectedSigningCABundle || (features.UseCacertsForSelfSignedCA && istioGenerated) {
```

Comment:
why remove the check for `UseCacertsForSelfSignedCA`

Reply:
Moved it to a separate case - easier to read and different log message. A bit of duplication - but more clarity.